src/adjoint/adjointAPI.F90

```diff
@@ -232,6 +235,11 @@ subroutine computeMatrixFreeProductBwdFast(dwbar, wbar, stateSize)
     ! of the fact that only wbar needs to be zeroed since all other
     ! required seeds are zeroed in the individual fast routines. This is
     ! slightly unsafe, but it necessary for speed.
+    
+    ! Allocate the memory for reverse
```

Comment:
is this only used for the debug functions? would it ever interfere with the regular use-cases?

Reply:
Yes, `solveAdjointForRHS` allocates the memory first during typical use. 
Since it checks that the memory is not allocated first there is no chance of double allocation.